src/TicketingSystem.BusinessLogic/Services/GenericEntityService.cs

```diff
@@ -0,0 +1,96 @@
+﻿using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using TicketingSystem.BusinessLogic.Dtos;
+using TicketingSystem.BusinessLogic.Exceptions;
+using TicketingSystem.DataAccess.Entities;
+using TicketingSystem.DataAccess.Repositories;
+
+namespace TicketingSystem.BusinessLogic.Services
+{
+    public class GenericEntityService<TEntity, TEntityDto>
+        where TEntity : class, IHasId
+        where TEntityDto : class, IDto
+    {
+        private protected readonly IMongoRepository<TEntity> _repository;
+        private protected readonly IMapper _mapper;
+
+        public GenericEntityService(IMongoRepository<TEntity> repository, IMapper mapper)
+        {
+            _repository = repository;
+            _mapper = mapper;
+        }
+
+        public Task CreateAsync(TEntityDto entity, CancellationToken cancellationToken = default)
+        {
+            return _repository.CreateAsync(_mapper.Map<TEntity>(entity), cancellationToken);
+        }
+
+        public async Task<IReadOnlyCollection<TEntityDto>> GetAllAsync(CancellationToken cancellationToken = default)
+        {
+            return _mapper.Map<List<TEntityDto>>(
+                    await _repository.GetAllAsync(cancellationToken));
+        }
+
+        public async Task<TEntityDto> GetByIdAsync(string entityId, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                return _mapper.Map<TEntityDto>(
+                    await _repository.GetByIdAsync(entityId, cancellationToken));
+            }
+            catch (ArgumentException)
```

Comment:
Why we use try-catch in some methods while do not in others? And also what if some other than `ArgumentException` will occur?